### DISCUSSION TITLE
blog/the-dynamic-features-of-java

### DISCUSSION THREAD

**Discussion by giscus** [cesarsotovalero/cesarsotovalero.github.io#68] [Announcements]
# blog/the-dynamic-features-of-java

The existence of dynamic features built in the language allows Java developers to transform their program executions at runtime dynamically. However, these features in most Java programs are a fundamental problem for static analysis tools that rely on precise call-graph construction. Notably, the GraalVM compiler relies on points-to analysis to perform AOT compilation. This blog post covers the main dynamic features of Java and the reasons why they are still a long-standing issue for researchers and practitioners in program analysis.

https://www.cesarsotovalero.net/blog/the-dynamic-features-of-java.html

**Comment by algomaster99** [comment]
Hello, Cesar! I might have a possible correction for you to make, or I did not interpret the text correctly.

> Now, we have to create a new .cpp file for the implementation of the execute function.

Given the example snippet, isn't `sayHello` the native function? If yes, `sayHello` should be implemented in the `C++` code rather than `execute`.

**Reply by cesarsotovalero** [reply]
Hi @algomaster99, thanks for your comment and the remark. You're right! I should have defined `sayHello()` in the C++ code, not in the `execute()` function. I have edited the text in the corresponding section 😄.

**Reply by algomaster99** [reply]
Thanks!
